src/backend/optimizer/util/clauses.c

```diff
@@ -5128,6 +5140,9 @@ evaluate_expr(Expr *expr, Oid result_type, int32 result_typmod,
 	/* Release all the junk we just created */
 	FreeExecutorState(estate);
 
+	if (sql_dialect == SQL_DIALECT_TSQL && pltsql_trunc_numeric_result_hook && getBaseType(result_type) == NUMERICOID)
```

Comment:
should only be doing this if const is not null?

Reply:
Added null check